src/mono/wasm/build/WasmApp.Common.targets

```diff
@@ -499,12 +499,22 @@
     <ItemGroup>
       <_BoolPropertiesThatTriggerRelinking Include="InvariantTimezone" DefaultValueInRuntimePack="false" />
       <_BoolPropertiesThatTriggerRelinking Include="InvariantGlobalization" DefaultValueInRuntimePack="false" />
+      <_BoolPropertiesThatTriggerRelinking Include="WasmNativeStrip" DefaultValueInRuntimePack="true" />
+      <_BoolPropertiesThatTriggerRelinking Include="WasmSingleFileBundle" DefaultValueInRuntimePack="false" />
+      <_WasmChangedProperty Include="%(_BoolPropertiesThatTriggerRelinking.Identity)" Condition="'$(%(_BoolPropertiesThatTriggerRelinking.Identity))' != '' and 
+                                                                                      '$(%(_BoolPropertiesThatTriggerRelinking.Identity))' != '%(_BoolPropertiesThatTriggerRelinking.DefaultValueInRuntimePack)'" />
     </ItemGroup>
 
     <PropertyGroup>
-      <WasmBuildNative Condition="'$(WasmBuildNative)' == '' and
-                          '$(%(_BoolPropertiesThatTriggerRelinking.Identity))' != '' and
-                          '$(%(_BoolPropertiesThatTriggerRelinking.Identity))' != '%(_BoolPropertiesThatTriggerRelinking.DefaultValueInRuntimePack)'">true</WasmBuildNative>
+      <_WasmBuildNativeRequired Condition="'$(%(_BoolPropertiesThatTriggerRelinking.Identity))' != '' and 
+                                '$(%(_BoolPropertiesThatTriggerRelinking.Identity))' != '%(_BoolPropertiesThatTriggerRelinking.DefaultValueInRuntimePack)'">true</_WasmBuildNativeRequired>
+    </PropertyGroup>
+
+    <Error Condition="'$(WasmBuildNative)' == 'false' and '$(_WasmBuildNativeRequired)' == 'true'" 
+           Text="Wasm native build is necessary. The value of %(_WasmChangedProperty.Identity) has been modified." />
```

Comment:
The message isn't very clear on what happened here. Maybe something like:
`WasmBuildNative is required because %24(%(_WasmChangedProperty.Identity))=$(%(_WasmChangedProperty.Identity)), but %24(WasmBuildNative) is already set to 'false'.`.